### PR TITLE
Update initial cpp unittests

### PR DESCRIPTION
**Pull request by gigony**
Update existing unit test code to use generated test image and allow external image specified by the parameter.

Updating `run` script to pre-generate the necessary test images and running C++ unit tests would be set up in a separate PR.

### CODE REVIEW

File: benchmarks/config.h [hunk 1]
```diff
@@ -42,6 +43,38 @@ struct AppConfig
     std::string benchmark_color; //  {auto|true|false}
     std::string benchmark_counters_tabular;
     std::string v; // <verbosity>
+
+    std::string get_input_path(const char* default_value = "generated/tiff_stripe_4096x4096_256.tif") const
```
Comment (jakirkham) [inline benchmarks/config.h:47]: Should `default_value` be `std::string`? Similar lines below

File: benchmarks/config.h [hunk 2]
```diff
@@ -21,7 +21,8 @@
 
 struct AppConfig
 {
-    std::string input_file = "test_data/private/generic_tiff_000.tif";
+    std::string test_folder;
+    std::string test_file;
```
Comment (jakirkham) [inline benchmarks/config.h:25]: We may want to revisit at some point and look at using [`std::filesystem::path`]( https://en.cppreference.com/w/cpp/filesystem/path ). Though that requires C++17 and we were already using `std::string`. So just a future note

**Review by jakirkham — COMMENTED** [review]
Thanks Gigon! 😄

Had a couple suggestions above

File: benchmarks/config.h (hunk 1 above)
Comment (gigony) [reply benchmarks/config.h]: Thanks! Updated to `const std::string`

File: benchmarks/config.h (hunk 2 above)
Comment (gigony) [reply benchmarks/config.h:25]: Thanks!
Created a issue for following up:
- https://github.com/rapidsai/cucim/issues/136

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
@gpucibot merge

**Pull request merged**